clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1190,6 +1195,15 @@ void handleResponse(AbstractResponse abstractResponse) {
             }
 
             @Override
+            boolean handleUnsupportedVersionException(UnsupportedVersionException exception) {
+                if (supportsDisablingTopicCreation) {
+                    supportsDisablingTopicCreation = false;
```

Comment:
I think that would still work in that case. We only ever need to fall back once: if the highest version of metadata request supported by the broker is v3 or lower. The first attempt will fail and the subsequent one will work.

To make it more concrete, say we introduce metadata request v5. If the broker supports v4 or v5, then it will just work. If the broker only supports v3, it's the same as now: we fail on the first attempt and then we fallback to not using the flag. Does that make sense?